protocol/message.go

```diff
@@ -118,6 +201,12 @@ func NewKeyLookupInEpochProof(ap *m.AuthenticationPath,
 	}, e
 }
 
+// NewMonitoringProof creates the response message a CONIKS server sends
+// to a client upon a MonitoringRequest. The contained DirectoryProofs
+// includes a list of proofs of inclusion and a list
+// of STRs for the epoch range [startEpoch, endEpoch], where startEpoch
+// and endEpoch are the epoch range endpoints indicated in the client's
+// request. The error code is set to Success.
```

Comment:
> "The error code is set to Success." <- That's confusing. But also in combination of what the constructor is doing: One can pass it an ErrorCode as a parameter (can be Success or anything else)

If you look at `directory.Monitor()`, which is the only place where `NewMonitoringProof` is called, the error code is set to `Success`, so that's what I was referring to. But you're definitely right that `ErrorCode` *could* be set to anything. I think it would make sense to changing the signature of the constructor and to set the `ErrorCode` internally to `Success` if that's the only "error" the constructor can ever return.

> This ErrorCode e is incorporated into the freshly created Response (OK) and returned (why)?

I had the same question a while back. The error is returned so it can be propagated up to the server for logging: https://github.com/coniks-sys/coniks-go/pull/71#r75061038